magento/pwa-studio
# [PWA-2190] [bug] - Fix stale cartId on multiple tabs

## Description

This PR fixes the problem of having staled cartId tokens saved on Redux when running the same store in multiple tabs.

In order to keep the cartId updated across tabs, this fix introduces a storage listener on cart context that fires a page reload whenever cartId is changed from another tab. The reload will force Redux to persist new cartId on storage.

 

## Related Issue







Closes PWA-2190

## Acceptance




Users should be able to run multiple tabs without having staled cartId saved on storage.

### Verification Stakeholders



### Specification




## Verification Steps






#### Test scenario(s) for direct fix/feature





1. Add  a product to cart
2. Open a new tab
3. Place an order in a new tab
4. Open first tab
5. Click add product to cart

- Verify if product is added to cart without any errors
- Verify automated test coverage is added for this use case


#### Test scenario(s) for any existing impacted features/areas






#### Test scenario(s) for any Magento Backend Supported Configurations





#### Is Browser/Device testing needed?





#### Any ad-hoc/edge case scenarios that need to be considered?




## Screenshots / Screen Captures (if appropriate)

## Breaking Changes (if any)





## Checklist
